### ISSUE TITLE
How can i using [sup + consistency or sup only]

### ISSUE THREAD

**Issue by whatseob** [LeeDoYup/FixMatch-pytorch#19]
Hello, I just studying Fixmatch, and wondering about this [fully supervised learning (50000 labels, sup only), (50000 labels, sup+consistency).
How can i chage or using it?
what is the default set when I using [--num_labels 50000]?

thanks

**Comment by LeeDoYup** [comment]
When you use [--num_labels 50000], the setting corresponds to the [50000 labels, sup+consistency] setting.
This repository assumes that the consistency regularization is always used, even though the weight of the consistency loss is zero. That is, the unlabeled data is always inferenced during training.
If you want to get the result of only supervised learning, you can re-write the training loop based on my code.

**Comment by whatseob** [comment]
Thanks for your reply! :)

May I ask one another question?
When I use  [--num_labels 4000], then I'm using 4000 labeled data and
[50000 unlabeled data or 46000 unlabeled data] ?
Because we have 50000 data, and  if i use 4000 labels, and i think maybe
46000 unlabeled data makes sense.. but i don't know exactly.. sorry


2021년 12월 14일 (화) 오후 6:08, Doyup Lee ***@***.***>님이 작성:

**Comment by LeeDoYup** [comment]
@Seob95 
Although we have 4000 labeled samples, the consistency regularization uses all 50000 samples. The implementation is same with the original implementation. It is because that there is no reason to exclude the labeled samples for the consistency regularization, since it helps the training of both labeled and unlabeled samples.

**Comment by whatseob** [comment]
Thank you for the clarification!! :)
